### PR TITLE
Use local clusterState call during healthchecks

### PR DESCRIPTION
**Pull request by gargharsh3134**
### Description

During health checks, while trying to get the optimised health checks property for all the nodes, a cluster state call is made to Opensearch which eventually is served by the ClusterManager node and hence stresses it out. For large clusters having high number of indices and shards, the cluster state size grows and serving cluster state requests from each node starts to degrade the cluster manager's performance. The attribute for each node can very well be served from local cluster state of the node. As per cluster manager's functioning, a node whose cluster state is lagging will be kicked out of the cluster after 90 seconds of waiting. So, serving clusterState from local for OSD process should not lead to any functionality change, except a little time lag in case the cluster state is lagging on the local node. 

This change just updates the clusterState call to use the ```local=true``` query param, so that the request doesn't impact ClusterManager node.

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by ruanyl — DISMISSED** [review]

File: src/core/server/opensearch/version_check/ensure_opensearch_version.ts [hunk 1]
```diff
@@ -76,6 +76,7 @@ export const getNodeId = async (
      */
     const state = (await internalClient.cluster.state({
       metric: 'nodes',
+      local: true,
```
Comment (ruanyl) [inline src/core/server/opensearch/version_check/ensure_opensearch_version.ts:79]: Looking at the comment on this function call, it seems calling cluster manager node is by intention. Shall we adjust the comment if using `local: true` is what we want?

**Comment by ruanyl** [comment]
Could you please mark this PR to be ready to review? 
The code change looks fine to me, but I left a question about this old comment:
```
    /*
     * Using _cluster/state/nodes to retrieve the cluster_id of each node from cluster manager node which
     * is considered to be a lightweight operation to aggegrate different cluster_ids from the OpenSearch nodes.
     */
```

File: src/core/server/opensearch/version_check/ensure_opensearch_version.ts (hunk 1 above)
Comment (gargharsh3134) [reply src/core/server/opensearch/version_check/ensure_opensearch_version.ts:79]: Ack, will update the code comment. Thanks for pointing it out.

**Review by ashwin-pc — COMMENTED** [review]
Nice! Can you add a changeset to the PR. I think this change is worth calling out in the changelog. Will approve once thats ready.

**Comment by gargharsh3134** [comment]
> Nice! Can you add a changeset to the PR. I think this change is worth calling out in the changelog. Will approve once thats ready.

@ashwin-pc Thanks for taking a look. I have added the changelog fragment, please take a look again.

**Comment by gargharsh3134** [comment]
> Could you please mark this PR to be ready to review? The code change looks fine to me, but I left a question about this old comment:
> 
> ```
>     /*
>      * Using _cluster/state/nodes to retrieve the cluster_id of each node from cluster manager node which
>      * is considered to be a lightweight operation to aggegrate different cluster_ids from the OpenSearch nodes.
>      */
> ```

@ruanyl Thanks for looking into it. I have updated the code comment, please check again.

**Comment by ruanyl** [comment]
@gargharsh3134 While I was diving into the original implementation, I found the related issue of it: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/330

I'm not familiar with the context of it, would you mind to take a look and see if your changes are compatible with it?

**Comment by gargharsh3134** [comment]
> @gargharsh3134 While I was diving into the original implementation, I found the related issue of it: #330
> 
> I'm not familiar with the context of it, would you mind to take a look and see if your changes are compatible with it?

@ruanyl I went through the linked issue, and looks like the new changes should maintain compatibility. The issue calls out the problem of fanning out the health check calls directly to all the nodes, and incase some of the nodes are slow and under resource constraint, they can bring down the OSD process. The proposed solution was to just make a single _cluster/state call to figure out when to really fan out these health checks to all the nodes, instead of doing it each time.

In the existing implementation, the request path was OSD making a call to local OS process which then is calling clusterManager node, once clusterManager responds back to local OS, the same response is served to OSD. With the new changes, the only difference is, instead of local node calling cluster manager, it would serve the request from it's own local cluster state (which might be lagging if the node is slow but would eventually catch up).

**Review by ashwin-pc — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Pull request merged**